core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -114,6 +108,7 @@ class HIPSpace {
 
  private:
   int m_device;  ///< Which HIP device
+  hipStream_t m_stream;
```

Comment:
Just wondering why you would not explore this:
```suggestion
  HIP exec;
```
since `HIP` knows the device and stream already.

Reply:
Because we want to use this class in the `HIP` execution space constructor.

Reply:
It just feels like a lot of the `HIP`, `HIPSpace` and `HIPInternal` types (same for `Cuda` I guess) are extracting the stream and device ID themselves.

If Kokkos wants to strongly couple device ID and stream, it might make sense that have them together in one of these objects, right ?